File: Compra.java
import java.time.LocalDate;
import java.util.ArrayList;

public class Compra {
    private static int nextID = 1;
    private int id;
    private ArrayList<Producto> productos = new ArrayList<>();
    private double total;
    private LocalDate fechaDeCompra;
    private Cliente cliente;
    
    public Compra(ArrayList<Producto> productos, double total, Cliente cliente) {
        this.productos = productos;
        this.total = total;
        this.id = nextID;
        nextID++;
        this.fechaDeCompra = LocalDate.now();
        this.cliente = cliente;
    }

    public void imprimirListaDeCompras(){
        System.out.println("Compra #" + id);
        System.out.println("  * Lista de productos:");
        for (int i = 0; i < productos.size(); i++) {
            Producto producto = productos.get(i);
            System.out.println(String.format("     * %s \t $%.2f", producto.getNombre(), producto.getPrecio()));
        }
    }

    public void imprimirInfoCompra(){
        System.out.println(String.format("Compra %d\t Total: $%.2f\t Cliente: %s", id, total, cliente.getNombre()));
    }

    public int getID(){
        return id;
    }

    public double getTotal() {
        return total;
    }
    
}

File: Compra.py
class Compra:
    nextID = 1

    def __init__(self, productos, total, cliente):
        self.id = Compra.nextID
        Compra.nextID += 1
        self.productos = productos
        self.total = total
        self.cliente = cliente

    def imprimirListaDeCompras(self):
        print("Compra #" + str(self.id))
        print("  * Lista de productos:")
        for producto in self.productos:
            print("     * {} \t ${:.2f}".format(producto.getNombre(), producto.getPrecio()))

    def imprimirInfoCompra(self):
        print("Compra {}\t Total: ${:.2f}\t Cliente: {}".format(self.id, self.total, self.cliente.getNombre()))

    def getID(self):
        return self.id

    def getTotal(self):
        return self.total